src/batch/src/executor/join/lookup_join.rs

```diff
@@ -313,7 +341,9 @@ impl<P: 'static + ProbeSideSourceBuilder> LookupJoinExecutor<P> {
 
             self.probe_side_source.reset();
             for row_key in &row_keys {
-                self.probe_side_source.add_scan_range(row_key)?;
+                if row_key.iter().all(|datum| datum.is_some()) {
+                    self.probe_side_source.add_scan_range(row_key)?;
```

Comment:
Correct me if I am wrong,
if there is no `add_scan_range` that gets called here, we can just move on to the next loop. Then we don't need a `DummyExecutor`?

Reply:
If the join is left outer join or left anti join, then we still need to add the row into the final join result. By using a `DummyExecutor`, we can reuse the existing join logic to handle this instead of making a special case for it.